samples/client/petstore/rust/src/apis/get_pet_by_id_api.rs

```diff
@@ -0,0 +1,23 @@
+use hyper;
+use serde_json;
+use futures::{Future, Stream};
+
+use super::models;
+use super::Error;
+
+pub fn get_pet_by_id<C: hyper::client::Connect>(
+    prefix: &str,
+    cli: &hyper::client::Client<C>,
+    pet_id: i64,
+    ) -> Box<Future<Item = models::Pet, Error = Error>> {
```

Comment:
(Here and it other generated methods)
Would you consider returning a more complex type? The Item here is losing any additional information about the response (header, status, etc) which I've found useful in some situations (e.g. token authenticated endpoints redirecting to token provider). For reference, reqwest does this with a richer `Response` [struct + trait](https://github.com/seanmonstar/reqwest/blob/8296c5e28e00a09ce45e0e33f8b10a8d15105a90/src/response.rs#L27).